src/db.rs

```diff
@@ -59,28 +60,40 @@ use crate::sst_iter::{SstIterator, SstIteratorOptions};
 use crate::tablestore::TableStore;
 use crate::types::{RowAttributes, ValueDeletable};
 use tracing::{info, warn};
+use uuid::Uuid;
 
 pub(crate) type FlushSender = tokio::sync::oneshot::Sender<Result<(), SlateDBError>>;
 pub(crate) type FlushMsg<T> = (Option<FlushSender>, T);
 
+pub(crate) type CheckpointSender =
```

Comment:
why create a different channel for this? I guess because you need to have a different result type? I think it might be better to just have one input channel for each task so then there's no ambiguities about message handling order (though I don't have a specific use case in mind). You could do that with a `ResultType` enum, though you lose some compile-time guarantees about the result type.

Reply:
I'll give it a shot and see how it looks. A bit unfortunate that we lose the binding between the send and receive types, but maybe it's not a big deal.